core/src/container.cpp

```diff
@@ -355,19 +395,51 @@ void SerialContainer::onNewSolution(const SolutionBase& current) {
 				solution.insert(solution.end(), out.first.begin(), out.first.end());
 				// store solution in sorted list
 				sorted.insert(std::make_shared<SolutionSequence>(std::move(solution), prio.cost(), this));
-			} else if (prio.depth() > 1) {
-				// update state priorities at both ends of newly created partial solution
-				updateStatePrio((in.first.empty() ? current : *in.first.back()).start(), prio);
-				updateStatePrio((out.first.empty() ? current : *out.first.back()).end(), prio);
+			}
+			if (prio.depth() > 1) {
+				// update state priorities along the whole partial solution path
+				updateStatePrio(current.start(), prio);
+				updateStatePrio(current.end(), prio);
+				updateStatePrios<Interface::BACKWARD>(in.first, prio);
+				updateStatePrios<Interface::FORWARD>(out.first, prio);
 			}
 		}
 	}
+	// printChildrenInterfaces(*this, true, *current.creator());
 
-	// finally store + announce new solutions to external interface
+	// finally, store + announce new solutions to external interface
 	for (const auto& solution : sorted)
 		impl->liftSolution(solution, solution->internalStart(), solution->internalEnd());
 }
 
+void SerialContainer::onNewFailure(const Stage& child, const InterfaceState* from, const InterfaceState* to) {
+	switch (child.pimpl()->interfaceFlags()) {
+		case GENERATE:
+			// just ignore: the pair of (new) states isn't known to us anyway
+			// TODO: If child is a container, from and to might have associated solutions already!
```

Comment:
I'm not sure what you mean here. If the child is a generator container, there are still no neighboring states we could prune.
Do you want to prune potential alternative branches *inside* the container?
That might indeed make sense for structured serial containers. (e.g., pick-connect-place).

Reply:
Today, it's too late to think this through again...

Reply:
Nobody except yourself expects you to respond right away :smile: